Components/Sensors/ThermocoupleTask.cpp

```diff
@@ -0,0 +1,365 @@
+
+	if(!(dataBuffer1[1] & 0x01)){ //if there is not an error with TC1 compute temperature
+		if((dataBuffer1[0]&(0x80))>>7==1)  // Negative Temperature (check if sign bit is 1)
+		{
+			//or together the 2 parts of the temperature after multiplying to the correct position
+			//if there is no error read all bits in buffer[0] and [1] will be temperature values
+			Temp1 = (dataBuffer1[0] << 6) | (dataBuffer1[1] >> 2);
+
+			//since the temperature is negative we need to do 2's compliment calculations
+			Temp1^=0b11111111111111; //first XOR all 14 bits of temp data including the sign bit (bits 31-18)
+									 //this will flip the bits
+			Temp1+=0b1; //then add 1
+
+			//here we first divide by 4 as the lower 2 bits are decimal bits, then because of this we scale
+			//the number to fit in an int_16t so it includes the decimal digits and we
+			//also correct the temperature by 3.2C which was the approximate error recorded
+			temperature1 = (((double)-Temp1 / 4)*100-320);
+		}
+		else  // Positive Temperature
+		{
+			//or together the 2 parts of the temperature after multiplying to the correct position
+			//if there is no error read all bits in buffer[0] and [1] will be temperature values
+			Temp1 = (dataBuffer1[0] << 6) | (dataBuffer1[1] >> 2);
+
+			//here we scale the number to fit in an int_16t so it includes the decimal digits and we
+			//also correct the temperature by 3.2C which was the approximate error recorded
+			temperature1 = (((double)Temp1 / 4)*100-320);
+		}
+	}
+	else
+	{
+		temperature1 = ERROR_TEMPURATURE_VALUE; //there is an error detected with TC1
+	}
+
+
+
+	tempDataBuffer5[5] = {0}; //reset for TC2 reading
+	//See Above bit mem-map
+
+	//Read ---------------------------------------------------------------------------------------
+	HAL_GPIO_WritePin(TC1_CS__GPIO_Port, TC1_CS__Pin, GPIO_PIN_SET);
+	HAL_GPIO_WritePin(TC2_CS__GPIO_Port, TC2_CS__Pin, GPIO_PIN_SET);
+
+	//Read From Thermocouple 1 first
+	HAL_GPIO_WritePin(TC1_CS__GPIO_Port, TC1_CS__Pin, GPIO_PIN_RESET); //begin read with CS pin low
+	HAL_Delay(10);
+	HAL_SPI_Receive(SystemHandles::SPI_Thermocouple1, tempDataBuffer5, 5, 1000); //Fill the data buffer with data from TC1
+	HAL_Delay(10);
+	HAL_GPIO_WritePin(TC2_CS__GPIO_Port, TC2_CS__Pin, GPIO_PIN_SET); //end read with setting CS pin to high again
+
+	for(int i = 0; i<5; i++){
+		if(i!=4)
+		dataBuffer2[i] = tempDataBuffer5[i];
+	}
+
+	int Temp2=0;
+
+	if(!(dataBuffer2[1] & 0x01)){ //if there is not an error with TC1 compute temperature
+		if((dataBuffer2[0]&(0x80))>>7==1)  // Negative Temperature
+		{
+			Temp2 = (dataBuffer2[0] << 6) | (dataBuffer2[1] >> 2);
+			Temp2^=0b11111111111111;
+			Temp2+=0b1;
+			temperature2 = (((double)-Temp2 / 4)*100-320);
+		}
+		else  // Positive Temperature
+		{
+			Temp2 = (dataBuffer2[0] << 6) | (dataBuffer2[1] >> 2);
+			temperature2 = (((double)Temp2 / 4)*100-320);
+		}
+	}
+	else
+	{
+		temperature2 = ERROR_TEMPURATURE_VALUE; //there is an error detected with TC2
+	}
+}
```

Comment:
You're again doing the same thing twice here. This should be generalized and extracted into a separate function that is called for each thermocouple.